src/relations/db.py

```diff
@@ -207,31 +204,40 @@ def set_up_relation(self, relation: Relation) -> bool:
                 )
             )
 
+            postgresql_version = None
+            try:
+                postgresql_version = self.charm.postgresql.get_postgresql_version()
+            except PostgreSQLGetPostgreSQLVersionError:
+                logger.exception(
+                    "Failed to retrieve the PostgreSQL version to initialise/update %s relation"
+                    % self.relation_name
+                )
+
```

Comment:
K8S charm will not update the endpoints as aggressively as VM. This will be reran only on config change. Should I try add something to update status?

Reply:
If it's possible to avoid calls to `sleep` or similar things on the notice event received by the charm (if we need to notify again from inside that event handler because the database version couldn't still be retrieved), I'm fine with using Pebble notices for that. Otherwise, I'd keep it on update status for now.